include/mesh/distributed_mesh.h

```diff
@@ -117,7 +117,13 @@ class DistributedMesh : public UnstructuredMesh
    * Virtual copy-constructor, creates a copy of this mesh
    */
   virtual std::unique_ptr<MeshBase> clone () const override
-  { return std::make_unique<DistributedMesh>(*this); }
+  {
+    auto returnval = std::make_unique<DistributedMesh>(*this);
+#ifdef DEBUG
+    libmesh_assert(*returnval == *this);
```

Comment:
It took me a while to realize that `MeshBase::operator==` eventually calls the virtual `MeshBase::subclass_locally_equals()` function. It might be worth updating the docs for `MeshBase::operator==` to explain that this non-virtual function internally calls a customized equivalence comparison which is appropriate to the 
underlying type of `other_mesh`.